plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -956,47 +956,51 @@ private RemoteTableName constructPageSinkIdsTable(ConnectorSession session, Conn
 
     @Override
     public void finishInsertTable(ConnectorSession session, JdbcOutputTableHandle handle, Set<Long> pageSinkIds)
+    {
+        RemoteTableName targetTable = new RemoteTableName(
+                Optional.ofNullable(handle.getCatalogName()),
+                Optional.ofNullable(handle.getSchemaName()),
+                handle.getTableName());
+        String columns = handle.getColumnNames().stream()
+                .map(this::quoted)
+                .collect(joining(", "));
+
+        // last args will be handled in the finish operation
+        String insertSql = "INSERT INTO %s (%s)".formatted(postProcessInsertTableNameClause(session, quoted(targetTable)), columns) + " %s";
```

Comment:
instead of appending a `%s` eagerly here, maybe the `finishOperation` should append the last arg itself.